packages/providers/upload-aws-s3/lib/index.js

```diff
@@ -10,19 +10,25 @@ const _ = require('lodash');
 const AWS = require('aws-sdk');
 
 module.exports = {
-  init(config) {
+  init({baseUrl = null, rootPath = null, s3Options, ...legacyS3Options}) {
+    if(legacyS3Options && process.env.NODE_ENV !== 'production')
+      console.log("S3 configuration options passed at root level of the plugin's providerOptions is deprecated and will be removed in a future release. You wrap them inside the 's3Options:{}' property.")
+
     const S3 = new AWS.S3({
       apiVersion: '2006-03-01',
-      ...config,
+      ...s3Options,
+      ...legacyS3Options
     });
 
     const upload = (file, customParams = {}) =>
       new Promise((resolve, reject) => {
         // upload file on S3 bucket
         const path = file.path ? `${file.path}/` : '';
+        const fileKey = `${rootPath ? rootPath.replace(/\/+$/, "") + "/" : ""}${path}${file.hash}${file.ext}`
```

Comment:
`${rootPath ? rootPath.replace(/\/+$/, "") + "/" : ""}` part could be applied already within `init` function.